pkg/updatestrategy/aws_asg.go

```diff
@@ -316,26 +319,40 @@ func (n *ASGNodePoolsBackend) Terminate(node *Node, decrementDesired bool) error
 	}
 
 	_, err := n.asgClient.TerminateInstanceInAutoScalingGroup(params)
```

Comment:
We should still handle this `err` somehow. If it fails, debugging this would be very hard.

Reply:
It's handled down here after checking the status first: https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/61/files#diff-41f826bed623fac059b975e38370daffR348

Reply:
Right, I've missed that. It's very weird that we retry `instanceStatus` for potentially a very long time and then suddenly return an error that happened days ago in a completely different code, would you mind either refactoring it somehow or adding a comment?

Reply:
Refactored it to be more clear. Now, on error, it will try to get the state (but not care about it) and only return an error if there's also an error on getting the current instance state.

Otherwise it will just go into the backoff waiting for the instance to be terminated.